test/simulation/tests-security.go

```diff
@@ -0,0 +1,470 @@
+	ht.assert.NoError(err)
+	ht.assert.Len(res.InternalMatches, 0)
+	ht.assert.Len(res.SwapSuccesses, 0)
+	ht.assert.Len(res.SwapFailures, 1)
+	ht.assert.Equal(res.SwapFailures[0].FailureReason, "SwapTimedOut")
+	ht.assert.NotNil(res.RemainingOrder)
+	ht.assert.Equal(res.RemainingOrder.GetLocalId(), bobOrderReq.OrderId)
+	ht.assert.Equal(res.RemainingOrder.Quantity, bobOrderReq.Quantity)
+
+	// Alice had shutdown so we'll get an error.
+	e := <-aliceSwapFailuresChan
+	ht.assert.EqualError(e.err, "rpc error: code = Unavailable desc = transport is closing")
+
+	<-net.Alice.ProcessExit
+
+	// Cleanup.
+
+	removalRes, err := net.Bob.Client.RemoveOrder(ht.ctx, &xudrpc.RemoveOrderRequest{OrderId: bobOrderReq.OrderId})
+	ht.assert.NoError(err)
+	ht.assert.NotNil(removalRes)
+	ht.assert.Equal(removalRes.QuantityOnHold, uint64(0))
+
+	onchainFeesThreshold := int64(20000)
+
+	// Closing maker BTC channel and checking balance.
+
+	err = closeBtcChannel(ht.ctx, net.LndBtcNetwork, net.Alice.LndBtcNode, aliceBtcChanPoint, false)
+	ht.assert.NoError(err)
+
+	aliceBalance, err := getBalance(ht.ctx, net.Alice)
+	ht.assert.NoError(err)
+	walletDiff := alicePrevBalance.btc.wallet.TotalBalance - aliceBalance.btc.wallet.TotalBalance
+	ht.assert.True(walletDiff < onchainFeesThreshold,
+		"alice btc wallet balance mismatch (prev: %v, current: %v)", alicePrevBalance.btc.wallet.TotalBalance, aliceBalance.btc.wallet.TotalBalance)
+
+	// Closing taker LTC channel and checking balance.
+	//
+	// First, find the pending HTLC expiration height.
+
+	bobLtcChanList, err := net.Bob.LndLtcNode.ListChannels(ht.ctx, &lnrpc.ListChannelsRequest{})
+	ht.assert.Equal(len(bobLtcChanList.Channels), 1)
+	bobChan := bobLtcChanList.Channels[0]
+	ht.assert.True(bobChan.RemoteBalance == 0)
+	ht.assert.True(bobChan.LocalBalance > 0)
+	ht.assert.Equal(len(bobChan.PendingHtlcs), 1)
+	expirationHeight := bobChan.PendingHtlcs[0].ExpirationHeight
+
+	// Try to expire the HTLC.
+
+	ltcInfo, err := net.LndLtcNetwork.LtcMiner.Node.GetInfo()
+	ht.assert.NoError(err)
+	_, err = net.LndLtcNetwork.LtcMiner.Node.Generate(expirationHeight - uint32(ltcInfo.Blocks) + 100)
+	ht.assert.NoError(err)
+	ltcInfo, err = net.LndLtcNetwork.LtcMiner.Node.GetInfo()
+	ht.assert.True(ltcInfo.Blocks > int32(expirationHeight))
+
+	err = net.Bob.LndLtcNode.WaitForBlockchainSync(ht.ctx)
+	ht.assert.NoError(err)
+	err = net.Alice.LndLtcNode.WaitForBlockchainSync(ht.ctx)
+	ht.assert.NoError(err)
+
+	// Attempt to close the channel cooperatively.
+
+	err = closeLtcChannel(ht.ctx, net.LndLtcNetwork, net.Bob.LndLtcNode, bobLtcChanPoint, false)
+	ht.assert.EqualError(err, "rpc error: code = Unknown desc = cannot co-op close channel with active htlcs")
+
+	// Close the channel uncooperatively.
+
+	err = closeLtcChannel(ht.ctx, net.LndLtcNetwork, net.Bob.LndLtcNode, bobLtcChanPoint, true)
+	ht.assert.NoError(err)
+
+	//_, err = net.LndLtcNetwork.LtcMiner.Node.Generate(1100)
```

Comment:
Should these lines be removed? I figure this is where you're trying to see if the balance will show up in lnd, but ultimately it shouldn't be needed because it should show up in the unconfirmed and total balance numbers immediately.

Reply:
Yes, it was now removed.